src/main/java/wellnus/storage/ReflectionTokenizer.java

```diff
@@ -0,0 +1,110 @@
+package wellnus.storage;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+
+import wellnus.exception.TokenizerException;
+
+/**
+ * Class to tokenize and detokenize the Index for 'like' and 'pref' command in Reflection Feature. <br>
+ */
+public class ReflectionTokenizer implements Tokenizer<Set<Integer>> {
+    private static final String INDEX_DELIMITER = ",";
+    private static final int INDEX_ZERO = 0;
+    private static final int INDEX_ONE = 1;
+    private static final String LIKE_KEY = "like";
+    private static final String PREF_KEY = "pref";
+    private static final String DETOKENIZE_ERROR_MESSAGE = "Detokenization failed!"
+            + "The file might be corrupted";
+    private String splitParameter(String tokenizedRawString, String parameterKey) throws TokenizerException {
+        int i = tokenizedRawString.indexOf(' ');
+        String parameter = tokenizedRawString.substring(INDEX_ZERO, i);
+        if (!parameter.equals(parameterKey)) {
+            throw new TokenizerException(DETOKENIZE_ERROR_MESSAGE);
+        }
+        String tokenizedIndexes = tokenizedRawString.substring(i).trim();
+        return tokenizedIndexes;
+    }
+    private String[] splitTokenizedIndex(String tokenizedIndexes) {
+        tokenizedIndexes = tokenizedIndexes.strip();
+        int noLimit = -1;
+        String[] rawString = tokenizedIndexes.split(INDEX_DELIMITER, noLimit);
+        String[] cleanString = new String[rawString.length];
+        for (int i = 0; i < rawString.length; ++i) {
+            String currentCommand = rawString[i];
+            currentCommand = currentCommand.strip();
+            cleanString[i] = currentCommand;
+        }
+        return cleanString;
+    }
+    private Set<Integer> getSet(String indexToSplit) throws TokenizerException {
+        Set<Integer> cleanIndexes = new HashSet<>();
```

Comment:
is ```indexSet``` or something like this a better naming? As this set actually serves the purpose of containing indexes but it is not supposed to be clean/empty

Reply:
`outputIndexes` would be better, since this is a plural data structure like arrays